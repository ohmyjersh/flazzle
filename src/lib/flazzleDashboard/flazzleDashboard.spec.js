import React from 'react';
import { shallow, mount } from 'enzyme';
import FlazzleDashboard from './index';

test('renders correctly', () => {
    const wrapper = shallow(<FlazzleDashboard title={"thing stuff"} flags={{flag:true,flag1:false}} goBack={() => {}} updateFlags={() => {}}  />);
    expect(wrapper).toMatchSnapshot();
});

test('renders correctly', () => {
    let updateFn = jest.fn();
    let backFn = jest.fn();
    const wrapper = mount(<FlazzleDashboard title={"thing stuff"} flags={{}} goBack={backFn} updateFlags={updateFn}  />);
    let saveButton = wrapper.find('button').findWhere(x => x.props().title === 'Save');
    saveButton.simulate('submit');
    // expect(experimental).toBeCalled();
    expect(updateFn.mock.calls.length).toBe(1);
    // expect(stable.mock.calls.length).toBe(0);
});

test('close button does what it should', () => {
    let updateFn = jest.fn();
    let backFn = jest.fn();
    const wrapper = mount(<FlazzleDashboard title={"thing stuff"} flags={{flag:true,flag1:false}} goBack={backFn} updateFlags={updateFn}  />);
    let closeButton = wrapper.find('button').findWhere(x => x.props().title === 'Close');
    closeButton.simulate('click');
    expect(backFn.mock.calls.length).toBe(1);
    expect(updateFn.mock.calls.length).toBe(0);
});