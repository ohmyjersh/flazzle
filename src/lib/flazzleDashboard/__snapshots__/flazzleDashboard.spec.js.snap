// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlazzleDashboard
    flags={
        Object {
            "flag": true,
            "flag1": false,
          }
    }
    goBack={[Function]}
    title="thing stuff"
    updateFlags={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onChange={[Function]}
        onSubmit={[Function]}
>
        <header
                className="dashboardHeader"
        >
                <button
                        className="headerButton"
                        title="Save"
                        type="submit"
                >
                        Save
                </button>
                <span
                        className="featureHeader"
                >
                        thing stuff - 
                        Features
                </span>
                <button
                        className="headerButton"
                        onClick={[Function]}
                        title="Close"
                        type="button"
                >
                        Close
                </button>
        </header>
        <section
                className="featureList"
        >
                <Feature
                        enabled={true}
                        feature="flag"
                />
                <Feature
                        enabled={false}
                        feature="flag1"
                />
        </section>
</form>,
      "className": "dashboard",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="dashboardHeader"
>
            <button
                        className="headerButton"
                        title="Save"
                        type="submit"
            >
                        Save
            </button>
            <span
                        className="featureHeader"
            >
                        thing stuff - 
                        Features
            </span>
            <button
                        className="headerButton"
                        onClick={[Function]}
                        title="Close"
                        type="button"
            >
                        Close
            </button>
</header>,
          <section
            className="featureList"
>
            <Feature
                        enabled={true}
                        feature="flag"
            />
            <Feature
                        enabled={false}
                        feature="flag1"
            />
</section>,
        ],
        "onChange": [Function],
        "onSubmit": [Function],
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="headerButton"
                title="Save"
                type="submit"
>
                Save
</button>,
              <span
                className="featureHeader"
>
                thing stuff - 
                Features
</span>,
              <button
                className="headerButton"
                onClick={[Function]}
                title="Close"
                type="button"
>
                Close
</button>,
            ],
            "className": "dashboardHeader",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save",
                "className": "headerButton",
                "title": "Save",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Save",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "thing stuff - ",
                  "Features",
                ],
                "className": "featureHeader",
              },
              "ref": null,
              "rendered": Array [
                "thing stuff - ",
                "Features",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Close",
                "className": "headerButton",
                "onClick": [Function],
                "title": "Close",
                "type": "button",
              },
              "ref": null,
              "rendered": "Close",
              "type": "button",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Feature
                enabled={true}
                feature="flag"
/>,
              <Feature
                enabled={false}
                feature="flag1"
/>,
            ],
            "className": "featureList",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "enabled": true,
                "feature": "flag",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "enabled": false,
                "feature": "flag1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "form",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onChange={[Function]}
          onSubmit={[Function]}
>
          <header
                    className="dashboardHeader"
          >
                    <button
                              className="headerButton"
                              title="Save"
                              type="submit"
                    >
                              Save
                    </button>
                    <span
                              className="featureHeader"
                    >
                              thing stuff - 
                              Features
                    </span>
                    <button
                              className="headerButton"
                              onClick={[Function]}
                              title="Close"
                              type="button"
                    >
                              Close
                    </button>
          </header>
          <section
                    className="featureList"
          >
                    <Feature
                              enabled={true}
                              feature="flag"
                    />
                    <Feature
                              enabled={false}
                              feature="flag1"
                    />
          </section>
</form>,
        "className": "dashboard",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="dashboardHeader"
>
              <button
                            className="headerButton"
                            title="Save"
                            type="submit"
              >
                            Save
              </button>
              <span
                            className="featureHeader"
              >
                            thing stuff - 
                            Features
              </span>
              <button
                            className="headerButton"
                            onClick={[Function]}
                            title="Close"
                            type="button"
              >
                            Close
              </button>
</header>,
            <section
              className="featureList"
>
              <Feature
                            enabled={true}
                            feature="flag"
              />
              <Feature
                            enabled={false}
                            feature="flag1"
              />
</section>,
          ],
          "onChange": [Function],
          "onSubmit": [Function],
        },
        "ref": [Function],
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="headerButton"
                  title="Save"
                  type="submit"
>
                  Save
</button>,
                <span
                  className="featureHeader"
>
                  thing stuff - 
                  Features
</span>,
                <button
                  className="headerButton"
                  onClick={[Function]}
                  title="Close"
                  type="button"
>
                  Close
</button>,
              ],
              "className": "dashboardHeader",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Save",
                  "className": "headerButton",
                  "title": "Save",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Save",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "thing stuff - ",
                    "Features",
                  ],
                  "className": "featureHeader",
                },
                "ref": null,
                "rendered": Array [
                  "thing stuff - ",
                  "Features",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Close",
                  "className": "headerButton",
                  "onClick": [Function],
                  "title": "Close",
                  "type": "button",
                },
                "ref": null,
                "rendered": "Close",
                "type": "button",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Feature
                  enabled={true}
                  feature="flag"
/>,
                <Feature
                  enabled={false}
                  feature="flag1"
/>,
              ],
              "className": "featureList",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "function",
                "props": Object {
                  "enabled": true,
                  "feature": "flag",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "function",
                "props": Object {
                  "enabled": false,
                  "feature": "flag1",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "section",
          },
        ],
        "type": "form",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
